cumulusci/tasks/bulkdata/update_data.py

```diff
@@ -0,0 +1,226 @@
+import csv
+from contextlib import contextmanager
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from snowfakery import generate_data
+
+from cumulusci.core.exceptions import BulkDataException, TaskOptionsError
+from cumulusci.core.utils import (
+    process_bool_arg,
+    process_list_arg,
+    process_list_of_pairs_dict_arg,
+)
+from cumulusci.tasks.bulkdata.step import (
+    DataApi,
+    DataOperationStatus,
+    DataOperationType,
+    get_dml_operation,
+    get_query_operation,
+)
+from cumulusci.tasks.bulkdata.utils import RowErrorChecker
+from cumulusci.tasks.salesforce import BaseSalesforceApiTask
+
+
+class UpdateData(BaseSalesforceApiTask):
+    """Update records of an sObject matching a where-clause."""
+
+    task_options = {
+        "object": {
+            "description": "An object",
+            "required": True,
+        },
+        "where": {
+            "description": "A SOQL where-clause (without the keyword WHERE).",
+            "required": False,
+        },
+        "api": {
+            "description": "The desired Salesforce API to use, which may be 'rest', 'bulk', or "
+            "'smart' to auto-select based on record volume. The default is 'smart'.",
+            "required": False,
+        },
+        "fields": {
+            "description": "Fields to download as input to the Snowfakery recipe",
+            "required": False,
+        },
+        "recipe": {
+            "description": "Snowfakery recipe to be executed on each row",
+            "required": True,
+        },
+        "recipe_options": {
+            "required": False,
+            "description": """Pass values to override options in the format VAR1:foo,VAR2:bar
+
+            Example: --recipe_options weight:10,color:purple""",
+        },
+        "ignore_row_errors": {
+            "description": "If True, allow the operation to continue even if individual rows fail to delete."
+        },
+    }
+    row_warning_limit = 10
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        self.sobject = self.options["object"]
+
+        def identifier(f):
+            return isinstance(f, str) and f.isalnum()
```

Comment:
Will `isalnum()` give false negatives if `object` is a custom object because the name of a custom object contains "__c"?

Reply:
Fixed now!